src/Platform/Microsoft.Testing.Platform/Logging/FileLoggerProvider.cs

```diff
@@ -73,15 +72,10 @@ public ILogger CreateLogger(string categoryName)
         => new FileLoggerCategory(FileLogger, categoryName);
 
     public void Dispose()
-        => FileLogger?.Dispose();
+        => FileLogger.Dispose();
 
 #if NETCOREAPP
-    public async ValueTask DisposeAsync()
-    {
-        if (FileLogger is not null)
-        {
-            await FileLogger.DisposeAsync();
-        }
-    }
+    public ValueTask DisposeAsync()
+        => FileLogger.DisposeAsync();
```

Comment:
I wouldn't be concerned about any perf benefits here and would favor preserving stack traces for exceptions here.

```suggestion
    public async ValueTask DisposeAsync()
        => await FileLogger.DisposeAsync();
```